docs/adding-new-languages.md

```diff
@@ -0,0 +1,20 @@
+# Adding new languages to Semantic
+
+This document exists to outline the process associated with adding a new language to Semantic. Though the Semantic authors have architected the library such that adding new languages and syntax [requires no changes to existing code](https://en.wikipedia.org/wiki/Expression_problem), adding support for a new language is a nontrivial amount of work. Those willing to take the plunge will probably need a degree of Haskell experience.
+
+## The procedure
+
```

Comment:
Maybe at the end of each bullet point describe which `semantic` subcommands will work at that point?